### PR TITLE
e2e-tests for MySQL plugins

### PR DESCRIPTION
**Pull request by poojantcs**
@Vipinofficial11 @priyabhatnagar25

### CODE REVIEW

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License..
+
+Feature: MySQL Sink - Design time scenarios
+
+  @TS-MYSQL-SINK-DSGN-01 @MYSQL_SOURCE_TEST
+  Scenario: Verify user can validate successfully when sink plugin is configured for table with basic details
```
Comment (MrRahulSharma) [inline mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature:19]: Here Source plugin steps are not required, please remove the unnecessary steps and perform the steps required for this scenario. Please update similar changes in other scenarios as well.

File: mysql-plugin/src/e2e-test/features/mysqlsource/RunTime.feature [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    And Save and Deploy Pipeline
+    And Run the Pipeline in Runtime
+    And Wait till pipeline is in running state
+    And Open and capture logs
+    And Verify the pipeline status is "Succeeded"
+    And Close the pipeline logs
+    Then Validate the values of records transferred to target table is equal to the values from source table
+
+  @TS-MYSQL-SOURCE-RNTM-02 @MYSQL_SOURCE_TEST @MYSQL_SINK_TEST @Mysql_Required
+  Scenario: To verify data is getting transferred from Mysql to Mysql successfully
```
Comment (Aryan-Verma) [inline mysql-plugin/src/e2e-test/features/mysqlsource/RunTime.feature]: We already have a Mysql to Mysql test. So please remove this

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    Then Replace input plugin property: "host" with value: "host" for Credentials and Authorization related fields
+    Then Replace input plugin property: "port" with value: "port" for Credentials and Authorization related fields
+    Then Replace input plugin property: "database" with value: "databaseName"
+    Then Replace input plugin property: "tableName" with value: "invalid.tablename"
+    Then Replace input plugin property: "user" with value: "username" for Credentials and Authorization related fields
+    Then Replace input plugin property: "password" with value: "password" for Credentials and Authorization related fields
+    Then Enter input plugin property: "referenceName" with value: "targetRef"
+    Then Validate "MySQL2" plugin properties
+    Then Verify that the Plugin is displaying an error message: "invalidtableName.error.message" on the header
+  @TS-MYSQL-SINK-DSGN-ERROR-03
```
Comment (Aryan-Verma) [inline mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature:82]: nit: add an empty line before

Comment (Aryan-Verma) [inline mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature]: nit: add an empty line before

File: mysql-plugin/src/e2e-test/features/mysqlsource/RunTime.feature (hunk 2 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsource/RunTime.feature]: Resolved

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature (hunk 1 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature:19]: Resolved

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature (hunk 3 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature]: Resolved

File: mysql-plugin/src/e2e-test/features/mysqlsource/MySql.feature [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright © 2022 Cask Data, Inc.
+# Copyright © 2023 Cask Data, Inc.
```
Comment (itsankit-google) [inline mysql-plugin/src/e2e-test/features/mysqlsource/MySql.feature:2]: why is this change required?

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License..
+
+Feature: MySQL Sink - Design time scenarios
+
+  @TS-MYSQL-SINK-DSGN-01 @MYSQL_SOURCE_TEST
```
Comment (MrRahulSharma) [inline mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature]: Why do we require the tags "TS-MYSQL-SINK-DSGN-**" ?

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License
+
+Feature: MySQL Sink - Design time validation scenarios
+
+  @TS-MYSQL-SINK-DSGN-ERROR-01
```
Comment (MrRahulSharma) [inline mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature]: Why do we require such tags ?

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+  Scenario: Verify Database field validation error message with invalid test data
```
Comment (MrRahulSharma) [inline mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature:19]: Source node is not required for this scenario, please remove the source node details from this scenario.

**Comment by MrRahulSharma** [comment]
Please ensure that we only configure the required node for testing in the validation tests. For example: We should ideally be using only Sink node to validate the config in the Sink and using only the source node to validate the config in the source.

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature (hunk 5 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature]: There is no such specific use ...Should we remove it?

Comment (MrRahulSharma) [reply mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature]: Yes.

File: mysql-plugin/src/e2e-test/features/mysqlsource/MySql.feature (hunk 4 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsource/MySql.feature:2]: Just Changed the current year in the license.

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature (hunk 5 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsink/DesignTime.feature]: Removed the tags.

File: mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature (hunk 7 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsink/DesignTimeValidation.feature:19]: So, here to get errors for invalid fields , we need to configure the source plugin also, otherwise we will not get error message for the invalid values entered in the sink plugin

File: mysql-plugin/src/e2e-test/features/mysqlsource/RunTime.feature [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+    And Verify the preview run status of pipeline in the logs is "succeeded"
+    And Close the pipeline logs
+    And Close the preview
+    And Save and Deploy Pipeline
+    And Run the Pipeline in Runtime
+    And Wait till pipeline is in running state
+    And Open and capture logs
+    And Verify the pipeline status is "Succeeded"
+    And Close the pipeline logs
+#    Then Validate the values of records transferred to target Big Query table is equal to the values from source table
```
Comment (Aryan-Verma) [inline mysql-plugin/src/e2e-test/features/mysqlsource/RunTime.feature]: Why is this condition commented out?

Comment (Vipinofficial11) [reply mysql-plugin/src/e2e-test/features/mysqlsource/RunTime.feature]: This Step is actually using a JDBC Driver for validation and that driver is not checked in cdap-e2e-tests for now so the Build is failing. That's why commented it to test the rest of the scenarios.

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/actions/MySQLPropertiesPageActions.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+package io.cdap.plugin.mysql.actions;
+
+import io.cdap.e2e.utils.ElementHelper;
+import io.cdap.e2e.utils.PluginPropertyUtils;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.plugin.mysql.locators.MySQLPropertiesPage;
+
+/**
+ * My SQL source - Properties page - Actions.
```
Comment (MrRahulSharma) [inline mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/actions/MySQLPropertiesPageActions.java]: Nitpik: Should be MySQL instead of My SQL.

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/locators/MySQLPropertiesPage.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+import io.cdap.e2e.utils.SeleniumDriver;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+import org.openqa.selenium.support.How;
+
+
+/**
+ * My SQL source - Properties page - Locators.
```
Comment (MrRahulSharma) [inline mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/locators/MySQLPropertiesPage.java]: Nitpik: Should be MySQL instead of My SQL.

File: mysql-plugin/src/e2e-test/features/mysqlsource/Datatypes.feature [hunk 11]
```diff
@@ -58,4 +59,42 @@ Feature: Mysql - Verify Mysql source data transfer for different datatypes
...
+    Then Click plugin property: "updateTableSchema"
+    Then Validate "BigQuery" plugin properties
+    Then Close the Plugin Properties page
+    Then Save the pipeline
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+#   Then Validate the values of records transferred to target Big Query table is equal to the values from source table
```
Comment (MrRahulSharma) [inline mysql-plugin/src/e2e-test/features/mysqlsource/Datatypes.feature]: Can we also implement this part of comparison of source and target table's data ?

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/actions/MySQLPropertiesPageActions.java (hunk 9 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/actions/MySQLPropertiesPageActions.java]: resolved

File: mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/locators/MySQLPropertiesPage.java (hunk 10 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/java/io/cdap/plugin/mysql/locators/MySQLPropertiesPage.java]: resolved

File: mysql-plugin/src/e2e-test/features/mysqlsource/Datatypes.feature (hunk 11 above)
Comment (priyabhatnagar25) [reply mysql-plugin/src/e2e-test/features/mysqlsource/Datatypes.feature]: Working on it.

**Review by Aryan-Verma — APPROVED** [review]

**Review by Nikitapaliwal123 — APPROVED** [review]

**Review by MrRahulSharma — APPROVED** [review]
